docs/IR.md

```diff
@@ -82,3 +81,32 @@ The following data types are supported by ONNX. Additional data types can be sup
 |Other|__bool__|Boolean value represent data with only two values, typically _true_ and _false_.|
 |Other|__handle__|Handles are opaque types holding a 64-bit integer.|
 |Collections|__sparse and dense tensor__|Tensors are a generalization of vectors and matrices; whereas vectors have one dimension, and matrices two, tensors can have any number of dimensions, including zero. A zero-dimensional tensor is equivalent to a scalar.|
+
+Extensibility
+-------------
+
+ONNX is expected to evolve over time and provides features that allow users to experiment and implement additions before they are folded into the specifications.
+
+### Metadata
+
+A model allows named metadata strings to be added via its `metadata_props` field, typically for use by tools such as converters, indexers or documentation generators. Names are not prescribed, but some name recommendations are provided for implementations that want to record such concepts.
+
+- `author`: name of the person(s) who authored the model
+- `company`: name of the company or organization that authored the model
+- `converted_from`: if converted from a different format, the name of the source format or framework
+- `license`: a human-readable name for a license if applicable
+- `license_url`: the URL where the license text is provided
+
+### Operators
+
+Operators may be extended via custom domain names in the `opset_import` field.
```

Comment:
maybe link to versioning doc or proto file directly? Or do we have a place that describes how opset_import works?

Reply:
That's described in Versioning.md but it really belongs here. I can do a followup where I pull in the non-versioning aspects of Versioning.md into IR.md; it's a bit odd that the mechanism for binding operators is described there and not here. But I don't feel particularly strong about it - do you have a preference?